src/java.base/share/classes/java/lang/String.java

```diff
@@ -4270,6 +4270,19 @@ public String translateEscapes() {
                 case 't':
                     ch = '\t';
                     break;
+                case 'u':
+                    if (from + 4 <= length) {
```

Comment:
Avoids a potential overflow
```suggestion
                    if (from <= length - 4) {
```

Reply:
Good.